sdcm/cluster.py

```diff
@@ -4385,6 +4389,34 @@ def stop_nemesis(self, timeout=10):
                 stack_trace = traceback.format_stack(current_thread_frames[nemesis_thread.ident])
                 threads_tracebacks.append("\n".join(stack_trace))
 
+    def start_kms_key_rotation_thread(self, rotation_interval_m=60) -> None:
```

Comment:
where this is coming from ? I don't remember we talk about introducing such a thread...

Reply:
I thought it would be a nemesis, like any other disruption we are doing
doing it on it's own thread meaning we have less information when it's happening (i.e. no events, like nemesis)

and if we want it in parallel to other nemesis, we cloud use parallel nemesis

Reply:
> I thought it would be a nemesis, like any other disruption we are doing doing it on it's own thread 

We never clarified the implementation of the rotation in our discussions. `I thought ` about a thread...

> meaning we have less information when it's happening (i.e. no events, like nemesis)

I can add an event creation here if really needed. For the moment it has a log message.

> and if we want it in parallel to other nemesis, we cloud use parallel nemesis

Why bother with nemesis? What is the gain?

Reply:
also we aren't doing the flow with calling `nodetool upgradesstable` in here at all.
and we probably want that case as well

Reply:
in a nemesis we have all the machinery in place, events, error handling, reporting, healthchecks.
and some degree of control on selecting them.

you are correctly is doesn't communicate directly with any specific node, so it's not the classic nemesis.

Reply:
> also we aren't doing the flow with calling `nodetool upgradesstable` in here at all. and we probably want that case as well

In the scope of the whole test run that longs for hours it is redundant.
The compactions, which happen all the time, will do the job we need - update sstables.